desk/sur/activity.hoon

```diff
@@ -0,0 +1,57 @@
+|%
++$  stream=(mop time event)
++$  indices=(map concern stream)
++$  concern
+  $%  [%group group-concern]
+      [%channel channel-concern]
+      [%dm dm-concern]
+      [%post post-concern]
+      [%writ writ-concern]
+  ==
++$  group-concern    group=flag
++$  channel-concern  [channel=nest group=flag]
++$  dm-concern       =whom
++$  post-concern     [=post-key channel=nest group=flag]
++$  writ-concern     [=writ-key =whom]
+::  $event: an instance of activity
+::
+::    $flavor: what activity generated the event and where did it happen
+::    $content: text of the message that generated the event or a description
+::    $level: what "level" of importance this is deemed by originator
+::    $read: has this event been seen or interacted with
+::
++$  event  [=flavor =level =content read=?]
++$  level  ?(%notify %default %trivial)
++$  content
+  $@  @t
+  $%  [%ship p=ship]
+      [%emph p=cord]
+  ==
++$  flavor
+  ::  specific occasions TBD
+  $%  [%group group-concern occasion=?(%join %kick)]
+      [%channel channel-concern occasion=?(%message %mention %reply %notice)]
+      [%dm dm-concern occasion=?(%message %mention %reply %notice)]
+      [%post post-concern occasion=?(%message %mention %reply %notice)]
+      [%writ writ-concern occasion=?(%message %mention %reply %notice)]
+  ==
```

Comment:
I suspect in practice these two types might be kind of coupled. Do we perhaps want per-occasion metadata instead of general `$content`? (Doesn't preclude us from using `$content` for the metadata where appropriate, like `%message`.)

Reply:
I think I miscopied from %hark, but content in event was supposed to be `content=(list content)` and used to generate the message like "x messaged you this: ....". Do you mean that different flavors might want other primitives in the content type?